### PR TITLE
fix(settings): cleanup connected account code

### PR DESCRIPTION
**Pull request by rohitkrai03**
Fixes: https://github.com/fabric8-services/fabric8-auth/issues/478

- Refresh OSO/Github account connection doesn't need to disconnect the account first. APIs for connect account handles that in the backend automatically. 
- Removed unnecessary connectGithub() call in refreshOpenShift() method.

### CODE REVIEW

**Comment by alien-ike** [comment]
### Ike Plugins (test-keeper)

Thank you @rohitkrai03 for this contribution!



It appears that no tests have been added or updated in this PR.

Automated tests give us confidence in shipping reliable software. Please add some as part of this change.

If you are an admin or the reviewer of this PR and you are sure that no test is needed then you can use the command `/ok-without-tests` as a comment to make the status green.


For more information please head over to official [documentation](http://arquillian.org/ike-prow-plugins/#_test_keeper_plugin). You can find there how to configure the plugin - for example exclude certain file types so if PR contains only them it won't be checked.

**Comment by jiekang** [comment]
I think this also addresses https://github.com/openshiftio/openshift.io/issues/3164 (possible duplicate issue)

**Comment by rohitkrai03** [comment]
@jiekang There are tests for this component but these changes doesn't affect any of them. And I've tested connectOpenShift for the implementation, it works as intended. If there is some edge case that I don't know about then I can't comment on that.

**Comment by rohitkrai03** [comment]
@jiekang I've added some more code that fixes another issue. Updated the description, take a look at it. If all looks good to you then we can merge this one today.

**Comment by fabric8cd** [comment]
@rohitkrai03 fabric8/fabric8-ui:SNAPSHOT-PR-2851-4 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2851-fabric8-ui.badger.fabric8.io

**Comment by jiekang** [comment]
The UX addition needs the UX team to look it over. Maybe it would be best to split it into a separate PR.

@catrobson @mceledonia Is this addition okay?

TLDR: If OpenShift is found to be disconnected, the issue is displayed to the user and a 'Connect' button now appears that allows users to re-connect the account. 

Hitting refresh also works but it's definitely not as obvious to the user, especially since before there would be no communication of the disconnected state. 

To clarify, what situation would cause a disconnected account to occur? @sbose78 @rohitkrai03 @alexeykazakov I think originally it was not an expected state so no button was included, but it looks like that was wrong. Is there a way I can test this easily?

**Comment by sbose78** [comment]
@jiekang 

You can disconnect openshift and test it out : 
```
curl -i -H "Authorization: Bearer $USER_TOKEN" -X DELETE https://auth.openshift.io/api/token?for=openshift
```

**Comment by jiekang** [comment]
May be best to do this with a different account.

1. Disconnect Openshift via `curl -i -H "Authorization: Bearer $USER_TOKEN" -X DELETE https://auth.openshift.io/api/token?for=openshift`

2. In local instance connected to production, visit settings page and see OpenShift disconnected with Connect Button available.

3. Click Connect Button.

Result: Redirected to `https://api.starter-us-east-2.openshift.com/oauth/authorize?access_type=online&client_id=openshift-io&redirect_uri=https%3A%2F%2Fauth.openshift.io%2Fapi%2Ftoken%2Flink%2Fcallback&response_type=code&scope=user%3Afull&state=8149a027-5d9f-4879-8c40-d3c25a353034`

with 'Authentication error occurred.'

Is this expected? It seems the connect process takes some time before the UI works again.

**Comment by jiekang** [comment]
Apart from that things work out.

**Comment by sbose78** [comment]
@jiekang did the reconnect work ? Authentication error could be because you are logged in to multiple RHDs. Did you try incognito?

**Comment by jiekang** [comment]
Yes reconnect worked for me. Just the redirect to auth error page was strange and it took some time.
